samples/browser/src/main/java/org/mozilla/samples/browser/BrowserFragment.kt

```diff
@@ -80,6 +87,33 @@ class BrowserFragment : BaseBrowserFragment(), UserInteractionHandler {
             view = layout
         )
 
+        fullScreenFeature.set(
+            feature = FullScreenFeature(
+                components.sessionManager,
+                SessionUseCases(components.sessionManager),
+                sessionId
+            ) { inFullScreen ->
+                if (inFullScreen) {
+                    // TODO how do I get access to the "focused" media being opened in
+                    //  full screen if it's not currently playing?
+                    // I tried adding a metadata "fullscreen" to onFullscreenChange
+                    // in Media but it's called after this callback
+                    val media =
+                        components.sessionManager.selectedSessionOrThrow.media.firstOrNull {
+                            it.state == Media.State.PLAYING
+                        }
+                    val height = media?.metadata?.height
+                    val width = media?.metadata?.width
+                    Logger("Height $height Width $width")
+                    activity?.enterToImmersiveMode()
```

Comment:
Following the breadcrumbs back to the Fenix issue... the goal here is to decide whether we should be in portrait mode or landscape mode when playing a video (depending on its aspect ratio)?

I wonder if another workaround (until we can actually know which `Media` we should look at) would be to not lock orientation in fullscreen at all. That is how RB does it and during fullscreen video playback you can still rotate the phone. This is still not perfect, but gives the user the option to choose (and if needed lock orientation changes globally).